### ISSUE TITLE
"Cannot read property 'downPointers' of undefined" when changing chart.data

### ISSUE THREAD

**Issue by RobbiesCode** [amcharts/amcharts4#983]
While trying to build a Column Chart Drill Down example I came accross this error message:

```
Cannot read property 'downPointers' of undefined
```

I am only replacing `data` with a new array (to show "drill down" items):

```
series.columns.template.events.on("hit", function(e){
  
  // update series datafields
  chart.series.getIndex(0).dataFields.valueY = "value";
  chart.series.getIndex(0).dataFields.categoryX = "category";
  
  var country = e.target.dataItem.dataContext.country;
  
  chart.data = sub_data[country];
  
});
```

Codepen demo: https://codepen.io/team/amcharts/pen/a501be0585aa103b3df5216b3932db98?editors=1010

**Comment by martynasma** [comment]
On click, you update `categoryX` data field of the series, but `category` data field of the category axis remains old one. Need to update it as well:

```
chart.xAxes.getIndex(0).dataFields.category = "category";
```

**Comment by RobbiesCode** [comment]
Of course, overlooked that completely. Thanks!

**Comment by RobbiesCode** [comment]
Actually, the error is still there, even after update the `category` of the X Axis. I think it has to do with the hover or hit event. The columns are replaced, while the mouse is still over the column...

```
// throws error
series.columns.template.events.on("hit", function(e){
  displayChildData();
});

// no error
series.columns.template.events.on("hit", function(e){
  setTimeout(displayChildData, 100);
});
```

**displayChildData** function:

```
  function displayChildData(){

    // update series datafields
    chart.series.getIndex(0).dataFields.valueY = "value";
    chart.series.getIndex(0).dataFields.categoryX = "category";

    categoryAxis.dataFields.category = "category";

    var country = e.target.dataItem.dataContext.country;

    chart.data = sub_data[country];
    
  }
```

**Comment by martynasma** [comment]
Fixed in [4.2.3](https://github.com/amcharts/amcharts4/releases/tag/4.2.3).

## [4.2.3] - 2019-03-14

### Fixed
- Exporting via API from a chart that did not have `ExportMenu` set, was resulting in error.
- Fixed tooltip flickering that were set directly on series objects (columns/bullets).
- Because of daylight saving, sometimes `DateAxis` could mistake yearly data for monthly data if `baseInterval` was not set explicitly.
- Sometimes Sankey nodes did not perfectly fit into chart area.
- Tick `location` on Y axis was being ignored.
- Ticks were positioned incorrectly when multiple axes were on the same side of the `XYChart`.
- Issue with `strictMinMax` and single data item fixed.
- `valueY` (or any other value) wasn't working in `legendSettings.itemLabelText`.

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
